Add application page for lake loader

istreeter opened the pull request.

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/_overview.md:
```diff
@@ -0,0 +1,33 @@
+  stream-->loader-->bucket
+`}/>
+
+<p>The lake loader is published as a Docker image which you can run on any {props.cloud} VM.  You do not need a Spark cluster to run this loader.</p>
```
This should not be in the same reusable block as the chart. The reason is that we then need to import the chart here: https://docs.snowplow.io/docs/storing-querying/loading-process/

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/_overview.md:
```diff
@@ -0,0 +1,33 @@
+
+<CodeBlock language="bash">{
+`docker run \\
+  -v/path/to/myconfig:/myconfig \\
```
I believe Docker now recommends using the `--mount type=bind,...` syntax (which is way more verbose but oh well)

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,78 @@
+---
+title: "Lake Loader (Delta)"
```
Do we need `(Delta)` in the title?

istreeter commented:
- Added the configuration reference
- Elaborated a bit more on the overview page
- Addressed earlier PR comments.

Ready for another review @stanch

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/configuration-reference-azure/index.md:
````diff
@@ -0,0 +1,84 @@
+import CodeBlock from '@theme/CodeBlock';
+```
+
+<p>The configuration reference in this page is written for Lake Loader <code>{`${versions.lakeLoader}`}</code></p>
````
I think I would prefer a single configuration page with tabs. Especially because we could have two independent groups of tabs: for the cloud and for the data format.

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,75 @@
+- If you set this to a **low** value, then you write events to the lake more frequently, but the output file size might be smaller than ideal.
+- If you set this to a **high** value, then you generate big output parquet files which are efficient for queries, but it adds a delay to events entering your lake.
+
+The default setting is `5 minutes` which is a nice balance between the need for large output parquet files and the need for reasonably low latency data.
```
Surely we can only make this claim for a given event throughput?

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,75 @@
+    "windowing": "5 minutes"
+}`}</CodeBlock>
+
+If you tune this setting correctly, then your lake can support efficient analytic queries without the need to run a `OPTIMIZE` job on the files.
```
```suggestion
If you tune this setting correctly, then your lake can support efficient analytic queries without the need to run an `OPTIMIZE` job on the files.
```

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,75 @@
+
+### Iglu
+
+The lake loader requires a [Iglu resolver file](/docs/pipeline-components-and-applications/iglu/iglu-resolver/index.md) which describes the Iglu repositories that host your schemas.  This should be the same Iglu configuration file that you used in the Enrichment process.
```
```suggestion
The lake loader requires an [Iglu resolver file](/docs/pipeline-components-and-applications/iglu/iglu-resolver/index.md) which describes the Iglu repositories that host your schemas.  This should be the same Iglu configuration file that you used in the Enrichment process.
```

istreeter commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/_deploy_overview.md:
```diff
@@ -0,0 +1,21 @@
+`docker run \\
+  --mount=type=bind,source=/path/to/myconfig,destination=/myconfig \\
+  snowplow/lake-loader-${props.cloud.toLowerCase()}:${versions.lakeLoader} \\
+  --config=/myconfig/loader.hocon \\
```
Like... mount the files into a known location and it auto-discovers them?

(I think Snowbridge recently made that change)

istreeter commented:
@stanch I think I addressed all your points.

However, this exercise has highlighted that the example config files in the lake-loader repo need some attention before they are appropriate for the documentation reference.  That's OK -- I need to do a lake-loader release this week anyway.

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,88 @@
+
+"Windowing" is an important config setting, which controls how often the lake loader commits a batch of events to the data lake. If you adjust this config setting, you should be aware that data lake queries are most efficient when the size of the parquet files in the lake are relatively large.
+
+- If you set this to a **low** value, then you write events to the lake more frequently, but the output file sizes will be smaller.
```
```suggestion
- If you set this to a **low** value, the loader will write events to the lake more frequently, reducing latency. However, the output parquet files will be smaller, which will make querying the data less efficient.
```

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,88 @@
+"Windowing" is an important config setting, which controls how often the lake loader commits a batch of events to the data lake. If you adjust this config setting, you should be aware that data lake queries are most efficient when the size of the parquet files in the lake are relatively large.
+
+- If you set this to a **low** value, then you write events to the lake more frequently, but the output file sizes will be smaller.
+- If you set this to a **high** value, then you generate bigger output parquet files which are efficient for queries, but it adds a delay to events entering your lake.
```
```suggestion
- Conversely, if you set this to a **high** value, the loader will generate bigger output parquet files, which are efficient for queries — at the cost of events arriving to the lake with more delay.
```

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/index.md:
```diff
@@ -0,0 +1,88 @@
+- If you set this to a **low** value, then you write events to the lake more frequently, but the output file sizes will be smaller.
+- If you set this to a **high** value, then you generate bigger output parquet files which are efficient for queries, but it adds a delay to events entering your lake.
+
+The default setting is `5 minutes`.  Above a certain event volume,  this default setting strikes a nice balance between the need for large output parquet files and the need for reasonably low latency data.
```
```suggestion
The default setting is `5 minutes`.  For moderate to high volumes, this value strikes a nice balance between the need for large output parquet files and the need for reasonably low latency data.
```

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/configuration-reference/index.md:
```diff
@@ -0,0 +1,103 @@
+
+You can use the `input.consumerConf` and `output.bad.producerConf` options to configure authentication to Azure event hubs using SASL.  For example:
+
+<CodeBlock>{`\
```
This can be a regular markdown code block (no need for escaping as well)

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/configuration-reference/index.md:
```diff
@@ -0,0 +1,103 @@
+  </TabItem>
+</Tabs>
+
+## Other processing config options
```
```suggestion
## Other configuration options
```

Many of them have nothing to do with processing

stanch commented on docs/pipeline-components-and-applications/loaders-storage-targets/lake-loader/configuration-reference/index.md:
```diff
@@ -0,0 +1,103 @@
+
+### Table configuration
+
+<Tabs groupId="format" queryString>
```
```suggestion
<Tabs groupId="lake-format" queryString>
```

A bit more descriptive so that we can use it on other pages throughout the docs without conflicts
